src/LibPQ.jl

```diff
@@ -619,52 +711,86 @@ function handle_result(jl_result::Result; throw_error::Bool=true)
+        conversions::Associative=LibPQ.PQConversions(),
+    ) -> Result
 
-Run a query on the PostgreSQL database and return a Result.
+Run a query on the PostgreSQL database and return a `Result`.
 If `throw_error` is `true`, throw an error and clear the result if the query results in a
 fatal error or unreadable response.
+
+The query may be passed as `Connection` and `AbstractString` (SQL) arguments, or as a
+`Statement`.
+
+`execute` optionally takes a `parameters` vector which passes query parameters as strings to
+PostgreSQL.
+
+`column_types` accepts type overrides for columns in the result which take priority over
+those in `type_map`.
+For information on the `column_types`, `type_map`, and `conversions` arguments, see
+[Type Conversions](@ref).
 """
-function execute(jl_conn::Connection, query::AbstractString; throw_error=true)
+function execute end
+
+function execute(jl_conn::Connection, query::AbstractString; throw_error=true, kwargs...)
     return handle_result(
-        Result(libpq_c.PQexec(jl_conn.conn, query));
+        Result(libpq_c.PQexec(jl_conn.conn, query), jl_conn; kwargs...);
         throw_error=throw_error,
     )
 end
 
-"""
-    execute(jl_conn::Connection, query::AbstractString, parameters::Vector{<:Parameter}; throw_error=true) -> Result
-
-Run a query on the PostgreSQL database and return a Result.
-If `throw_error` is `true`, throw an error and clear the result if the query results in a
-fatal error or unreadable response.
-"""
 function execute(
     jl_conn::Connection,
     query::AbstractString,
-    parameters::AbstractVector{<:Parameter};
+    parameters::AbstractVector;
     throw_error=true,
+    kwargs...
 )
     num_params = length(parameters)
+    string_params = string_parameters(parameters)
 
     return handle_result(
         Result(libpq_c.PQexecParams(
             jl_conn.conn,
             query,
             num_params,
             C_NULL,  # set paramTypes to C_NULL to have the server infer a type
-            parameter_pointers(parameters),
+            parameter_pointers(string_params),
             C_NULL,  # paramLengths is ignored for text format parameters
             zeros(Cint, num_params),  # all parameters in text format
             zero(Cint),  # return result in text format
-        ));
+        ), jl_conn, kwargs...);
         throw_error=throw_error,
     )
 end
 
-function parameter_pointers(
-    parameters::AbstractVector{<:Parameter},
-)
+"""
+    string_parameters(parameters::AbstractVector) -> Vector{Union{String, Missing}}
+
+Convert parameters to strings which can be passed to libpq, propagating `missing`.
+"""
+function string_parameters end
+
+string_parameters(parameters::AbstractVector{<:Parameter}) = parameters
+function string_parameters(parameters::AbstractVector{>:Missing})
+    map(parameters) do parameter
+        ismissing(parameter) ? parameter : string(parameter)
+    end
+end
+string_parameters(parameters::AbstractVector) = map(string, parameters)
```

Comment:
I'm feeling method claustrophobia. Maybe add some whitespace

Reply:
Is there an issue if you do `Any[missing]`?

Reply:
No issue with that, but there is an issue with `Any[missing, 3]`.